app/services/admin_service/workplace_search.rb

```diff
@@ -0,0 +1,110 @@
+class AdminService::WorkplaceSearch
+  attr_reader :q
+
+  def initialize(q:)
+    @offset = 0
+    @limit = 0
+    @q = q
+  end
+
+  def count(_)
+    scopes.inject(0) { |sum, scope| sum + scope.count }
+  end
+
+  def offset(offset)
+    @offset = offset
+    self
+  end
+
+  def limit(limit)
+    @limit = limit
+    self
+  end
+
+  def each(&blk)
+    find_scopes.each(&blk)
+  end
+
+private
+
+  def schools_scope
+    chain = School.order(name: :asc)
+    if q.present?
+      chain = chain.where(urn: q)
+      chain = chain.or(School.where("name ILIKE ?", "%#{q}%"))
+    end
+
+    chain
+  end
+
+  def local_authority_scope
+    chain = LocalAuthority.order(name: :asc)
+    if q.present?
+      chain = chain.where("name ILIKE ?", "%#{q}%")
+    end
+
+    chain
+  end
+
+  def private_childcare_provider_scope
+    chain = PrivateChildcareProvider.order(provider_name: :asc)
+    if q.present?
+      chain = chain.where(provider_urn: q)
+      chain = chain.or(PrivateChildcareProvider.where("provider_name ILIKE ?", "%#{q}%"))
+    end
+
+    chain
+  end
+
+  def scopes
+    [
+      schools_scope,
+      private_childcare_provider_scope,
+      local_authority_scope,
+    ]
+  end
+
+  # Return properly paginated scopes
+  def find_scopes
+    # Extra offset is to compensate for the previous scouts count.
```

Comment:
I'm not clear on what is meant by 'scouts count' thats not a term I'm familiar with